src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/MemoryCacheTest.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.AspNetCore.Razor.LanguageServer.Semantic.Services;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.Semantic
+{
+    public class MemoryCacheTest
```

Comment:
These all copied from the MemoryCache repo?

Reply:
Here's sort of the order of events here:
1. Tried to just use the MemoryCache assembly because DRY. There was a version mismatch issue because O# users MS.Extensions.Logging v2.0.0 and the normal MemoryCache version to use uses 5.0.
2. I started out aping the source in MemoryCache.cs but realized that our needs were vastly simpler than the complexity that class offers and ended up writing it mostly freehand, though the method signatures are similar.

Reading that again, are you asking about the MemoryCache tests or the class itself?

Reply:
> Reading that again, are you asking about the MemoryCache tests or the class itself?

The tests.

Oh and all that said, once we're able to use the memory cache assembly do you envision us removing all this?

Reply:
> Oh and all that said, once we're able to use the memory cache assembly do you envision us removing all this?
100%. Even though the MemoryCache I created is relatively simple I'm sure I've made mistakes/inefficiencies that don't happen in the more fleshed-out class, so we should switch to that when possible.